src/components/metadataSelects/useOptionsQuery.ts

```diff
@@ -0,0 +1,77 @@
+import { useEffect, useMemo, useRef, useState } from 'react'
+import { useModelGist } from '../../lib'
+import { SelectOption } from '../../types'
+import { CategoryComboQueryResult } from './types'
+
+export function useOptionsQuery({
+    initialSelected,
+    setSelectedOption,
+    fetchInitialOption,
+}: {
+    setSelectedOption: (option: SelectOption) => void
+    fetchInitialOption: () => void
+    initialSelected?: string
+}) {
+    const fetchedInitialOptionRef = useRef(false)
+    const [loadedOptions, setLoadedOptions] = useState<SelectOption[]>([])
+    const queryResult = useModelGist<CategoryComboQueryResult>(
+        'categoryCombos/gist',
+        // { pageSize: 10 },
+        {},
+        {
+            variables: { pageSize: 10, filter: '' },
+            onComplete: (input: unknown) => {
```

Comment:
I think it's worth noting that `react-query` are deprecating callbacks for queries in `v4`, and removing them in v5. But I think this is fine for now. I do things somewhat similar in `manageColumns` for the transfer with an `useEffect`. 

But since you already have an `useEffect` below, I think we could do it there?